Kudu.Tests/Core/Function/KafkaTriggerKedaAuthProviderTests.cs

```diff
@@ -0,0 +1,135 @@
+        {
+            KafkaTriggerKedaAuthProviderErrorMock kafkaTriggerKedaAuthProvider = new KafkaTriggerKedaAuthProviderErrorMock();
+            string jsonText = @"
+            {
+                ""protocol"": ""SaslSsl"",
+                ""authenticationMode"": ""Plain"",
+                ""username"": ""test"",
+                ""password"": ""test""
+            }";
+
+            JToken jsonObj = JToken.Parse(jsonText);
+            IDictionary<string, string> authRef = kafkaTriggerKedaAuthProvider.PopulateAuthenticationRef(jsonObj, "testFunctionName");
+            Assert.Null(authRef);
+        }
+
+        [Fact]
+        public void PopulateAuthenticationRef_Continues_When_AddSecretsFails()
+        {
+            KafkaTriggerKedaAuthProviderOverload kafkaTriggerKedaAuthProvider = new KafkaTriggerKedaAuthProviderOverload();
+            string jsonText = @"
+            {
+                ""protocol"": ""SaslSsl"",
+                ""authenticationMode"": ""Plain"",
+                ""username"": ""test"",
+                ""password"": ""test""
+            }";
+
+            JToken jsonObj = JToken.Parse(jsonText);
+            IDictionary<string, string> authRef = kafkaTriggerKedaAuthProvider.PopulateAuthenticationRef(jsonObj, "testFunctionName");
+            Assert.Equal(1, authRef.Count);
+        }
+
+        [Fact]
+        public void TestIFTriggerAuthIsNull_With_NoAuthenticationMode()
+        {
+            KafkaTriggerKedaAuthProviderOverload kafkaTriggerKedaAuthProvider = new KafkaTriggerKedaAuthProviderOverload();
+            string jsonText = @"
+            {
+                ""protocol"": ""SaslSsl"",
+                ""username"": ""test"",
+                ""password"": ""test""
+            }";
+
+            JToken jsonObj = JToken.Parse(jsonText);
+            IDictionary<string, string> authRef = kafkaTriggerKedaAuthProvider.PopulateAuthenticationRef(jsonObj, "testFunctionName");
+            Assert.Null(authRef);
+        }
+
+        [Fact]
+         public void TestIFTriggerAuthIsNull_With_NoProtocol()
+        {
+            KafkaTriggerKedaAuthProviderOverload kafkaTriggerKedaAuthProvider = new KafkaTriggerKedaAuthProviderOverload();
+            string jsonText = @"
+            {
+                ""authenticationMode"": ""Plain"",
+                ""username"": ""test"",
+                ""password"": ""test""
+            }";
+
+            JToken jsonObj = JToken.Parse(jsonText);
+            IDictionary<string, string> authRef = kafkaTriggerKedaAuthProvider.PopulateAuthenticationRef(jsonObj, "testFunctionName");
+            Assert.Null(authRef);
+        }
+
+
+        private class KafkaTriggerKedaAuthProviderOverload : KafkaTriggerKedaAuthProvider
+        {
+            internal override void CreateTriggerAuthenticationRef(IDictionary<string, string> secretKeyToKedaParam, string functionName)
+            {
+                // do nothing.
```

Comment:
Addressed with validating authRef data ->Assert.Equal(1, authRef.Count);

Reply:
I mean you are mocking with this class right? 
So that, usually we test if CreateTriggerAuthenticationRef method is properly called and get the expected parameters. If you use Moq library, it might be very easy. However, decision is up to you.

Reply:
Done. Added moq library and removed overload methods.